spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -0,0 +1,207 @@
+    return PARAMETERS;
+  }
+
+  @Override
+  public StructType outputType() {
+    return OUTPUT_TYPE;
+  }
+
+  @Override
+  public InternalRow[] call(InternalRow args) {
+    Identifier tableIdent = toIdentifier(args.getString(0), PARAMETERS[0].name());
+
+    return modifyIcebergTable(tableIdent, table -> {
+      RewriteDataFiles action = actions().rewriteDataFiles(table);
+
+      String strategy = args.isNullAt(1) ? null : args.getString(1);
+      String sortOrders = args.isNullAt(2) ? null : args.getString(2);
+      if (sortOrders != null && (strategy == null || strategy.equalsIgnoreCase("binpack"))) {
+        throw new IllegalArgumentException(
+                "sort_order is not accepted for binpack strategy. Please configure sort strategy to use sort_order");
+      }
+      SortOrder sortOrder = collectSortOrders(strategy, table, action, sortOrders);
+      action = checkAndApplyStrategy(action, strategy, sortOrder);
+
+      if (!args.isNullAt(3)) {
+        action = checkAndApplyOptions(args, action);
+      }
+
+      String where = args.isNullAt(4) ? null : args.getString(4);
+      action = checkAndApplyFilter(action, where);
+
+      RewriteDataFiles.Result result = action.execute();
+
+      return toOutputRows(result);
+    });
+  }
+
+  private RewriteDataFiles checkAndApplyFilter(RewriteDataFiles action, String where) {
+    if (where != null) {
+      ParserInterface sqlParser = spark().sessionState().sqlParser();
+      try {
+        Expression expression = sqlParser.parseExpression(where);
+        return action.filter(SparkExpressions.convert(expression));
+      } catch (ParseException e) {
+        throw new IllegalArgumentException("Cannot parse predicates in where option: " + where);
+      }
+    }
+    return action;
+  }
+
+  private RewriteDataFiles checkAndApplyOptions(InternalRow args, RewriteDataFiles action) {
+    Map<String, String> options = Maps.newHashMap();
+    args.getMap(3).foreach(DataTypes.StringType, DataTypes.StringType,
+            (k, v) -> {
+              options.put(k.toString(), v.toString());
+              return BoxedUnit.UNIT;
+            });
+    return action.options(options);
+  }
+
+  private RewriteDataFiles checkAndApplyStrategy(RewriteDataFiles action, String strategy, SortOrder sortOrder) {
+    if (strategy != null) {
+      if (strategy.equalsIgnoreCase("binpack")) {
+        return action.binPack();
+      } else if (strategy.equalsIgnoreCase("sort")) {
+        if (sortOrder == null) {
+          return action.sort();
+        }
+        return action.sort(sortOrder);
+      } else {
+        throw new IllegalArgumentException("unsupported strategy: " + strategy + ". Only binpack,sort is supported");
+      }
+    }
+    return action;
+  }
+
+  private SortOrder collectSortOrders(
+          String strategy, Table table, RewriteDataFiles action, String sortOrderStr) {
+    if (strategy != null && strategy.equalsIgnoreCase("sort") && sortOrderStr != null) {
```

Comment:
Instead of doing a custom parse code here is it possible that we can use the Spark Parser's parse ordered identifier list? If not I would add this to our own parser code and just use that rather than writing a manual parser here

Reply:
totally agree. 
we have a same syntax in `Alter table ... write ordered by ` in Iceberg and spark's `order by`
we should make use of it. 

I tried using `SqlBaseParser::orderedIdentifierList`,  but didn't work. I am new to this spark code.